4.student-exercises.js

```diff
@@ -0,0 +1,68 @@
+const getTrainersAndGymsAndPokemons = (gyms, trainers, pokemons) => {
+  // Combine trainers and pokemons and gyms
+  return (
+    trainers
+      // Map trainers to include their pokemons
+      .map(trainer => {
+        // Include the trainer's pokemons
+        trainer.pokemons = pokemons.filter(pokemon =>
```

Comment:
You can avoid this by:
```suggestion
        const trainerPokemons = pokemons.filter(pokemon =>
```